src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+            throw new DataProcessingException("Cannot add a book" + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(ID_INDEX, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            Book foundBook = null;
+            if (resultSet.next()) {
+                foundBook = resultSetFunction(resultSet);
+            }
+            return Optional.ofNullable(foundBook);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find a book with id:" + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> booksList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Book recievedBook = resultSetFunction(resultSet);
+                booksList.add(recievedBook);
+            }
+            return booksList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all data from DB", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(TITLE_POSITION, book.getTitle());
+            preparedStatement.setObject(PRICE_POSITION, book.getPrice());
+            preparedStatement.setObject(ID_POSITION, book.getId());
+            preparedStatement.executeUpdate();
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update such book: " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            preparedStatement.setLong(ID_INDEX, id);
+
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete book with id" + id, e);
+
+        }
+    }
+
+    private static Book resultSetFunction(ResultSet resultSet) {
```

Comment:
What exactly is this function? Rename it please to `parseBookFromResultSet` or just parseBook or getBook

Reply:
Thanks)
Also, I added a new line in Injector, but I still have the same error in Main class